src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -67,11 +67,12 @@ let rec untuple_args_pat typs (P_aux (paux, ((l, _) as annot)) as pat) =
 let doc_nexp (Nexp_aux (n, l) as nexp) =
   match n with Nexp_constant i -> string (Big_int.to_string i) | _ -> failwith "NExp not translatable yet."
 
-let doc_typ (Typ_aux (t, _) as typ) =
+let rec doc_typ (Typ_aux (t, _) as typ) =
   match t with
   | Typ_id (Id_aux (Id "unit", _)) -> string "Unit"
   | Typ_id (Id_aux (Id "int", _)) -> string "Int"
   | Typ_app (Id_aux (Id "bitvector", _), [A_aux (A_nexp m, _)]) -> string "BitVec " ^^ doc_nexp m
+  | Typ_tuple ts -> parens (separate (space ^^ string "×" ^^ space) (List.map doc_typ ts))
```

Comment:
There is a `separate_map` pretty printing combinator you can here, as it's a fairly common case